packages/utilities/src/utils/common.ts

```diff
@@ -25,54 +25,156 @@ export const extractUrlAndGlobal = (urlAndGlobal: string): [string, string] => {
-    }
-    // if its a function that must be called (lazily executing function)
-    else if (typeof value === 'function') {
-      // @ts-ignore
-      acc[key] = { asyncContainer: value };
-    }
-    // if its just a string (global@url)
-    else if (typeof value === 'string') {
-      const [url, global] = extractUrlAndGlobal(value);
-      acc[key] = { global, url };
-    }
-    // we dont know or currently support this type
-    else {
-      //@ts-ignore
-      console.log('remotes process', process.env.REMOTES);
-      throw new Error(
-        `[mf] Invalid value received for runtime_remote "${key}"`
-      );
-    }
-    return acc;
-  },
-  {} as RuntimeRemotesMap);
+  try {
+    //@ts-ignore
+    const remoteVars = (process.env.REMOTES || {}) as RemoteVars;
 
-  return runtimeRemotes;
+    const runtimeRemotes = Object.entries(remoteVars).reduce(function (
+      acc,
+      item
+    ) {
+      const [key, value] = item;
+      // if its an object with a thenable (eagerly executing function)
+      if (typeof value === 'object' && typeof value.then === 'function') {
+        acc[key] = { asyncContainer: value };
+      }
+      // if its a function that must be called (lazily executing function)
+      else if (typeof value === 'function') {
+        // @ts-ignore
+        acc[key] = { asyncContainer: value };
+      }
+      // if its just a string (global@url)
+      else if (typeof value === 'string') {
+        const [url, global] = extractUrlAndGlobal(value);
+        acc[key] = { global, url };
+      }
+      // we dont know or currently support this type
+      else {
+        //@ts-ignore
+        console.log('remotes process', process.env.REMOTES);
+        throw new Error(
+          `[mf] Invalid value received for runtime_remote "${key}"`
+        );
+      }
+      return acc;
+    },
+    {} as RuntimeRemotesMap);
+
+    return runtimeRemotes;
+  } catch (err) {
+    console.warn('Unable to retrieve runtime remotes: ', err);
+  }
+
+  return {} as RuntimeRemotesMap;
 };
 
-/**
- * Return initialized remote container by remote's key or its runtime remote item data.
- *
- * `runtimeRemoteItem` might be
- *    { global, url } - values obtained from webpack remotes option `global@url`
- * or
- *    { asyncContainer } - async container is a promise that resolves to the remote container
- */
-export const injectScript = (
+export const importDelegatedModule = async (
   keyOrRuntimeRemoteItem: string | RuntimeRemote
 ) => {
+  // @ts-ignore
+  return loadScript(keyOrRuntimeRemoteItem).then((asyncContainer) => {
+    // for legacy reasons, we must mark container a initialized
+    // here otherwise older promise based implementation will try to init again with diff object
+    // asyncContainer.__initialized = true;
+    return asyncContainer
+  }).then((asyncContainer) => {
+    // most of this is only needed because of legacy promise based implementation
+    if(typeof window === 'undefined') {
+      return asyncContainer
```

Comment:
proxy code is now moved up into the container loading system, should not need it here anymore